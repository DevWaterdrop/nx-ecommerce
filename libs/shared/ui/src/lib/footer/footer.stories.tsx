import { Story, Meta } from '@storybook/react';
import { Footer } from './footer';

export default {
  component: Footer,
  title: 'Components/Footer',
} as Meta;

const Template: Story = (args) => <Footer {...args} />;

export const Primary = Template.bind({});
