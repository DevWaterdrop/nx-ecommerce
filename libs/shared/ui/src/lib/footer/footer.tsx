import clsx from 'clsx';

export const Footer: React.FC = () => {
  return (
    <footer className={clsx('mt-20 w-full', 'bg-gray-100')}>
      <div
        className={clsx(
          'mx-auto max-w-screen-xl py-8 px-4',
          'flex gap-8 items-center justify-between'
        )}
      >
        <p className="text-base select-none">
          <span aria-label="flag of Ukraine" role="img">
            ðŸ‡ºðŸ‡¦
          </span>
          &nbsp;STOP WAR&nbsp;
          <span aria-label="flag of Ukraine" role="img">
            ðŸ‡ºðŸ‡¦
          </span>
        </p>
        <ul className={clsx('flex gap-4', 'text-xs not-italic', 'opacity-65')}>
          <li>
            <a
              className="hover:underline"
              rel="noreferrer noopener"
              target="_blank"
              href="https://github.com/DevWaterdrop/nx-ecommerce"
            >
              Github
            </a>
          </li>
          <li>
            <a
              className="hover:underline"
              rel="author noreferrer noopener"
              target="_blank"
              href="https://github.com/DevWaterdrop"
            >
              DevWaterdrop
            </a>
          </li>
        </ul>
      </div>
    </footer>
  );
};
