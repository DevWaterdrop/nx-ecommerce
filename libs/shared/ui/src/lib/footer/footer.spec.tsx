import { render, RenderResult, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import { Footer } from './footer';

describe('Footer', () => {
  let renderResult: RenderResult;

  beforeEach(() => {
    renderResult = render(<Footer />);
  });

  it('should render successfully', () => {
    const { baseElement } = renderResult;
    expect(baseElement).toBeTruthy();
  });

  it('should have author link', () => {
    expect(screen.getByRole('link', { name: 'DevWaterdrop' })).toBeTruthy();
  });
});
