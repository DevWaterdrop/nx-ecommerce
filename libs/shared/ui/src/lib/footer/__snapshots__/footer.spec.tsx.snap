// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render successfully 1`] = `
<footer
  className="mt-20 w-full bg-gray-100"
>
  <div
    className="mx-auto max-w-screen-xl py-8 px-4 flex gap-8 items-center justify-between"
  >
    <p
      className="text-base select-none"
    >
      <span
        aria-label="flag of Ukraine"
        role="img"
      >
        ðŸ‡ºðŸ‡¦
      </span>
      Â STOP WARÂ 
      <span
        aria-label="flag of Ukraine"
        role="img"
      >
        ðŸ‡ºðŸ‡¦
      </span>
    </p>
    <ul
      className="flex gap-4 text-xs not-italic opacity-65"
    >
      <li>
        <a
          className="hover:underline"
          href="https://github.com/DevWaterdrop/nx-ecommerce"
          rel="noreferrer noopener"
          target="_blank"
        >
          Github
        </a>
      </li>
      <li>
        <a
          className="hover:underline"
          href="https://github.com/DevWaterdrop"
          rel="author noreferrer noopener"
          target="_blank"
        >
          DevWaterdrop
        </a>
      </li>
    </ul>
  </div>
</footer>
`;
